doc/POINTS_COORDINATES.md

```diff
@@ -0,0 +1,169 @@
+# Points, tripoints, and coordinate systems
+
+## Coordinate systems
+
+CDDA uses a variety of coordinate systems for different purposes.  These differ
+by scale and origin.
+
+The most precise coordinates are **map square** (ms) coordinates.  These refer to
+the tiles you see normally when playing the game.
+
+Two origins for map square coordinates are common:
+* **Absolute** coordinates, which are a global system for the whole game,
+  relative to a fixed origin.
+* **Local** coordinates, which are relative to the corner of the current "reality
+  bubble", or `map` roughly centered on the avatar.  In local map square
+  coordinates, `x` and `y` values will both fall in the range [0,120).
+
+The next scale is **submap** (sm) coordinates.  One submap is 24x24
+(`SEEX`x`SEEY`) maps squares.  Submaps are the scale at which chunks of the map
+are loaded or saved as they enter or leave the reality bubble.
+
+Next comes **overmap terrain** (omt) coordinates.  One overmap terrain is 2x2
+submaps.  Overmap terrains correspond to a single tile on the map view in-game,
+and are the scale of chunk of mapgen.
+
+Largest are **overmap** (om) coordinates.  One overmap is 180x180
+(`OMAPX`x`OMAPY`) overmap terrains.  Large-scale mapgen (e.g. city layout)
+happens one overmap at a time.
+
+Lastly, these is a system called **segment** (seg) coordinates.  These are only
+used in saving/loading submaps and you are unlikely to encounter them.
+
+As well as absolute and local coordinates, sometimes we need to use coordinates
+relative so some larger scale.  For example, when performing mapgen for a
+single overmap, we want to work with coordinates within that overmap.  This
+will be an overmap terrain-scale point relative to the corner of its containing
+overmap, and so typically take `x` and `y` values in the range [0,180).
+
+## Vertical coordinates
+
+Although `x` and `y` coordinates work at all these various scales, `z`
+coordinates are consistent across all contexts.  They lie in the range
+[-`OVERMAP_DEPTH`,`OVERMAP_HEIGHT`].
+
+## Vehicle coordinates
+
+TODO.
```

Comment:
Vehicles use two map scales internally:
* **mount** scale is the same as map square, but is relative to a fixed point in the vehicle when the vehicle is facing due east.  the mount scale is fixed and unchanging, relative to the vehicle's (0, 0) point - with the minor complication that the (0, 0) point can move if the vehicle is damaged and all the vehicle parts at the location and destroyed.
* **vehicle** scale is also the same as map square, but translates the mount scale points along the vehicle's current facing (or some other axis).  A vehicle part's **ms** position is the **ms** position of the vehicle's (0, 0) **mount** scale part plus the vehicle part's **vehicle** scale position.

Vehicles *shear* as they change facing but do not *rotate*.  The logic to convert a **mount** point to a **vehicle** point is complicated and handled by the `vehicle::coord_translate()` and `vehicle::mount_to_tripoint()` families of functions.

Currently, vehicle **mount** co-ordinates do not have a z-level component, but vehicle **vehicle** coordinates do.  The `z` coordinate is relative to the *map scale* position of the vehicle's (0, 0) **mount** point.

Reply:
Thanks.  I added this, but I rewrote it quite a lot.  Let me know what you think.

Reply:
The new version looks good.  Thanks for improving it.